File: 560. Subarray Sum Equals K/hash.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
 var subarraySum = function(nums, k) {
   
    const freq = new Map();
    let res = 0;
    let sum = 0;
    freq.set(0, 1)
    for(const num of nums){
        sum += num;
        const diff = sum - k
        if(freq.has(diff)){
            res += freq.get(diff)
        }
        freq.set(sum, 1 + (freq.get(sum) || 0))
    }
    
    return res;
    
};

File: 0560. Subarray Sum Equals K/hash.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = total = 0;
        freq = {}
        freq[0] = 1;
        
        for num in nums:
            total += num;
            diff = total - k;
            
            if diff in freq:
                res += freq[diff];
            
            freq[total] = freq.get(total, 0) + 1;
        
        return res;
            